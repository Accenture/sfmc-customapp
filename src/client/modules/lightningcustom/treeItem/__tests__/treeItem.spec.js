/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import { shadowQuerySelector } from 'lightning/testUtils';
import Element from 'c/treeItem';

describe('c-tree-item', () => {
    it('renders leaf tree-item', () => {
        const element = createElement('c-tree-item', { is: Element });
        document.body.appendChild(element);
        element.isRoot = false;
        element.label = 'label';
        element.href = 'href';
        element.metatext = 'metatext';
        element.nodeRef = {
            expanded: false
        };

        element.isExpanded = false;
        element.isDisabled = false;
        element.nodename = '2234';
        element.nodeKey = '1.1';
        element.isLeaf = true;

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('uses the correct icon for LTR', () => {
        const element = createElement('c-tree-item', { is: Element });
        document.body.appendChild(element);

        element.isRoot = false;
        element.label = 'label';
        element.href = 'href';

        return Promise.resolve().then(() => {
            const lightningIcon = shadowQuerySelector(
                element,
                'c-primitive-icon'
            );

            expect(lightningIcon.iconName).toBe('utility:chevronright');
        });
    });

    it('uses the correct icon for RTL', () => {
        const element = createElement('c-tree-item', { is: Element });
        document.dir = 'rtl';
        document.body.appendChild(element);

        element.isRoot = false;
        element.label = 'label';
        element.href = 'href';

        return Promise.resolve().then(() => {
            const lightningIcon = shadowQuerySelector(
                element,
                'c-primitive-icon'
            );

            expect(lightningIcon.iconName).toBe('utility:chevronleft');
        });
    });
});
