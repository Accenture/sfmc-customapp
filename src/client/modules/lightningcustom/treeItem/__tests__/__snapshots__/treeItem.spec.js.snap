// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-tree-item renders leaf tree-item 1`] = `
<lightning-tree-item>
  #shadow-root(open)
    <div
      class="slds-tree__item"
    >
      <button
        aria-hidden="true"
        class="slds-button slds-button_icon slds-m-right_x-small slds-is-disabled"
        tabindex="-1"
        title="Expand Tree Branch"
        type="button"
      >
        <lightning-primitive-icon>
          #shadow-root(open)
            <svg
              aria-hidden="true"
              class="slds-button__icon slds-button__icon_small"
              data-key="chevronright"
              focusable="false"
            >
              <use
                xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
              />
            </svg>
        </lightning-primitive-icon>
        <span
          class="slds-assistive-text"
        >
          Expand Tree Branch
        </span>
      </button>
      <span
        class="slds-has-flexi-truncate"
      >
        <a
          href="href"
          role="presentation"
          tabindex="-1"
        >
          <span
            class="slds-tree__item-label slds-truncate"
            title="label"
          >
            label
          </span>
          <span
            class="slds-tree__item-meta slds-truncate"
            title="metatext"
          >
            <span
              class="slds-assistive-text"
            >
              :
            </span>
            metatext
          </span>
        </a>
      </span>
    </div>
</lightning-tree-item>
`;
